File: android/SalmonTrollingAssistant/app/src/main/java/com/example/salmontrollingassistant/domain/model/TideData.kt
package com.example.salmontrollingassistant.domain.model

import com.squareup.moshi.JsonClass
import java.util.Date
import java.util.UUID

enum class TideType {
    HIGH, LOW, RISING, FALLING
}

@JsonClass(generateAdapter = true)
data class TideEvent(
    val timestamp: Date,
    val height: Double
)

@JsonClass(generateAdapter = true)
data class TideData(
    val id: String = UUID.randomUUID().toString(),
    val timestamp: Date,
    val height: Double,
    val type: TideType,
    val nextHighTide: TideEvent? = null,
    val nextLowTide: TideEvent? = null
)

File: SalmonTrollingAssistant/Domain/Models/TideData.swift
import Foundation

struct TideData: Identifiable, Codable {
    enum TideType: String, Codable {
        case high
        case low
        case rising
        case falling
    }
    
    let id: String
    let timestamp: Date
    let height: Double
    let type: TideType
    
    init(id: String = UUID().uuidString, timestamp: Date = Date(), height: Double, type: TideType) {
        self.id = id
        self.timestamp = timestamp
        self.height = height
        self.type = type
    }
    
    static func mockData() -> TideData {
        TideData(
            timestamp: Date(),
            height: 3.2,
            type: .rising
        )
    }
}